framework/src/main/java/org/apache/drill/test/framework/DrillQueryProfile.java

```diff
@@ -0,0 +1,241 @@
+package org.apache.drill.test.framework;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonTypeName("queryProfile")
+public class DrillQueryProfile {
+
+    public String user;
+
+    @JsonProperty("start")
+    public long queryStartTime;
+
+    @JsonProperty("end")
+    public long queryEndTime;
+
+    @JsonProperty("foreman")
+    public Endpoint foreman;
+
+    @JsonProperty("state")
+    public String rootState;
+
+    public int totalFragments;
+
+    public int finishedFragments;
+
+    @JsonProperty("planEnd")
+    public long planEndTime;
+
+    @JsonProperty("queueWaitEnd")
+    public long queueWaitEndTime;
+
+    public long totalCost;
+
+    public String queueName;
+
+    public String queryId;
+
+    @JsonProperty("fragmentProfile")
+    public List<MajorFragmentProfile> fragmentProfiles;
+
+    @Override
+    public String toString() {
+        String ret = "DrillQueryProfile(" +
+                "user=\"" + user + "\", " +
+                "queryStartTime=" + queryStartTime + ", " +
+                "queryEndTime=" + queryEndTime + ", " +
+                "foreman=" + foreman + ", " +
+                "rootState=\"" + rootState + "\", " +
+                "totalFragments=" + totalFragments + ", " +
+                "finishedFragments=" + finishedFragments + ", " +
+                "planEndTime=" + planEndTime + ", " +
+                "queueWaitEndTime=" + queueWaitEndTime + ", " +
+                "totalCost=" + totalCost + ", " +
+                "queueName=\"" + queueName + "\", " +
+                "queryId=\"" + queryId + "\", ";
+        StringBuilder sb = new StringBuilder(ret);
+        if (fragmentProfiles != null) {
+            sb.append("fragmentProfile=[");
+
+            fragmentProfiles.forEach(frag -> sb.append(frag).append(", "));
+            if(fragmentProfiles.size() > 0) {
+                sb.deleteCharAt(sb.length() - 1).deleteCharAt(sb.length() - 1);
+            }
+            sb.append("])");
+        }
+
+        return sb.toString();
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    @JsonTypeName("endpoint")
+    public static class Endpoint {
+
+
```

Comment:
Remove blank lines and clean up formatting?

Reply:
Here and through the file

Reply:
Done.